ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -598,6 +627,16 @@ public void rollbackTxn() throws LockException {
     }
   }
 
+  @Override
+  public void rollbackTxn(Context ctx) throws LockException {
+    try {
+      cleanupOutputDir(ctx);
+    } catch (TException e) {
+      throw new LockException(ErrorMsg.METASTORE_COMMUNICATION_FAILED.getMsg(), e);
+    }
+    rollbackTxn();
```

Comment:
we should call rollback first, and then try the cleanup, if cleanup fails, we'll never mark a txn as aborted